gateway/gateway.go

```diff
@@ -146,6 +148,13 @@ func (g *gateway) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 		g.started = false
 	}
 
+	// capture C log output
+	loggingStarted, ok := LoggingRoutineStarted[g.Name().Name]
+	if !ok || !loggingStarted {
```

Comment:
what's the case where we're not okay, but the logging has not started?

If you want just a check for an element existing we can use a map[string]interface{} rather than a map[string]bool

Reply:
If the element does not exist in the map, but yeah in that case we can just change the map to a slice of names instead of a map.

Reply:
That is to say, we'll keep this map as it already is.